packages/sdk/src/types/Validations.ts

```diff
@@ -1,91 +1,486 @@
+	valid: boolean
+	applicable_to: ApplicableToObjectPromotionTier //6_res_applicable_to_object
+	inapplicable_to: InapplicableToObjectPromotionTier //6_res_inapplicable_to_object
+	tracking_id: string
+	order: ObjectOrderApplyToOrder //6_obj_order_object_apply_to_order
+	code: string
+	loyalty: {
+		points_cost: number
 	}
-	reward?: {
+	reward: {
 		id: string
+		assignment_id: string
+		points: number
 	}
-	session?: ValidationSessionParams
+	metadata: Record<string, any>
+	start_date: string
+	expiration_date: string
+	campaign: string
+	campaign_id: string
+	session: ValidationSessionParams
 }
 
-export interface ValidationsValidateVoucherResponse {
-	applicable_to?: ApplicableToResultList
-	inapplicable_to?: ApplicableToResultList
-	campaign?: string
-	campaign_id?: string
-	metadata?: Record<string, any>
-	code?: string
-	valid?: boolean
-	discount?: DiscountAmount | DiscountUnit | DiscountPercent | DiscountFixed
-	gift?: {
+interface ResponseValidateVoucherGiftCard {
+	valid: boolean
+	applicable_to: ApplicableToObjectPromotionTier //6_res_applicable_to_object
+	inapplicable_to: InapplicableToObjectPromotionTier //6_res_inapplicable_to_object
+	tracking_id: string
+	order: ObjectOrder
+	code: string
+	gift: {
 		amount: number
 		balance: number
+		effect: 'APPLY_TO_ORDER' | 'APPLY_TO_ITEMS'
 	}
-	loyalty?: {
-		points_cost: number
-	}
+	metadata: Record<string, any>
+	start_date: string
+	expiration_date: string
+	campaign: string
+	campaign_id: string
+	session: ValidationSessionParams
+}
+
+interface ResponseValidateVoucherDiscountCode {
+	valid: boolean
+	applicable_to: ApplicableToObjectPromotionTier //6_res_applicable_to_object
+	inapplicable_to: InapplicableToObjectPromotionTier //6_res_inapplicable_to_object
+	tracking_id: string
+	order: ObjectOrder
+	code: string
+	discount:
+		| ValidateVoucherDiscountAmount
+		| ValidateVoucherDiscountPercent
+		| ValidateVoucherDiscountFixed
+		| ValidateVoucherDiscountUnit
+		| ValidateVoucherDiscountUnitMultiple
+		| ValidateVoucherDiscountShipping
+	metadata: Record<string, any>
+	start_date: string
+	expiration_date: string
+	campaign: string
+	campaign_id: string
+	session: ValidationSessionParams
+}
+
+export interface ValidationsValidateStackableParams {
+	options?: StackableOptions
+	redeemables: (RedeemablesDiscountReferralPromotionTierPromotionStack | RedeemablesGiftCard | RedeemablesLoyaltyCard)[]
+	session?: ValidationSessionParams
+	order?: CreateOrder
+	customer?: CreateCustomer
+	metadata?: Record<string, any>
+}
+
+interface RedeemablesDiscountReferralPromotionTierPromotionStack {
+	type: 'voucher' | 'promotion_tier' | 'promotion_stack'
+	id: string
+}
+
+interface RedeemablesGiftCard {
+	id: string
+	object: 'voucher'
+	gift?: { credits: number }
+}
+
+interface RedeemablesLoyaltyCard {
+	id: string
+	object: 'voucher'
+	reward?: { id: string; points: number }
+}
+
+export type ValidationValidateStackableResponse = ResponseValidationsTrue | ResponseValidationsFalse
```

Comment:
Interface vs Type - please look at origin comment about the `DistributionsPublicationsCreateParams`